### PR TITLE
refactor(core): Extract axios utility helpers into axios-utils

### PR DESCRIPTION
**Pull request by tomi**
## Summary

PR 2/X to break request-helpers into more modular parts.

Continuation of https://github.com/n8n-io/n8n/pull/26852

Extracted 8 axios utility helper functions from the large `request-helper-functions.ts` (1877 lines) into a dedicated `axios-utils.ts` module to improve code organization and maintainability. The extracted functions handle URL validation, header management, request/response transformation, digest authentication, form data processing, and HTTP redirect handling.

New test suite added with 27 comprehensive unit tests covering all exported functions and their edge cases.

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/CAT-2581

## Review / Merge checklist

- [x] PR title and summary are descriptive
- [x] Tests included (27 new unit tests for all 8 extracted functions)
- [ ] Docs updated (not applicable - internal refactoring)

### CODE REVIEW

File: packages/core/src/execution-engine/node-execution-context/utils/request-helpers/axios-utils.ts [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+import crypto from 'crypto';
+import FormData from 'form-data';
+import type { AgentOptions } from 'https';
+import type { IHttpRequestOptions, IgnoreStatusErrorConfig } from 'n8n-workflow';
+
+import type { SsrfBridge } from '@/execution-engine';
+import { createHttpProxyAgent, createHttpsProxyAgent } from '@/http-proxy';
+
+/** Checks whether a string is a valid, parseable URL. */
+export function validateUrl(url?: string): boolean {
```
Comment (ivov) [inline packages/core/src/execution-engine/node-execution-context/utils/request-helpers/axios-utils.ts]: See comment in sibling PR about duplication of URL validation

File: packages/core/src/execution-engine/node-execution-context/utils/request-helper-functions.ts [hunk 2]
```diff
@@ -90,24 +99,9 @@ axios.defaults.paramsSerializer = (params) => {
...
-	return (
-		typeof ignoreHttpStatusErrors === 'object' &&
-		ignoreHttpStatusErrors !== null &&
-		'ignore' in ignoreHttpStatusErrors &&
-		ignoreHttpStatusErrors.ignore === true
-	);
-}
-
-function getUrlFromProxyConfig(proxyConfig: IHttpRequestOptions['proxy'] | string): string | null {
+export function getUrlFromProxyConfig(
```
Comment (ivov) [inline packages/core/src/execution-engine/node-execution-context/utils/request-helper-functions.ts]: Should we move this one too? Really odd that we're passing it as a 6th callback param to `getBeforeRedirectFn`

File: packages/core/src/execution-engine/node-execution-context/utils/request-helpers/axios-utils.ts (hunk 1 above)
Comment (tomi) [reply packages/core/src/execution-engine/node-execution-context/utils/request-helpers/axios-utils.ts]: I addressed this in the other PR. I think we need to align these two once either one of these is first in

File: packages/core/src/execution-engine/node-execution-context/utils/request-helpers/__tests__/axios-utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+	describe('validateUrl', () => {
+		test.each<[string | undefined, boolean]>([
+			['https://example.com', true],
+			['http://localhost:3000', true],
+			['ftp://files.example.com/path', true],
+			['not-a-url', false],
+			['://missing-protocol', false],
+			[undefined, false],
+			['', false],
+		])('should return %s for %p', (url, expected) => {
```
Comment (Flexicon) [inline packages/core/src/execution-engine/node-execution-context/utils/request-helpers/__tests__/axios-utils.test.ts]: nit: the arguments here are reversed I think, which will lead to output like: 
`should return https://example.com for true`
while `should return true for https://example.com` was probably intended.

Swapping around the args here would resolve the issue:

```suggestion
		test.each<[boolean, string | undefined]>([
			[true, 'https://example.com'],
			[true, 'http://localhost:3000'],
			[true, 'ftp://files.example.com/path'],
			[false, 'not-a-url'],
			[false, '://missing-protocol'],
			[false, undefined],
			[false, ''],
		])('should return %p for %s', (expected, url) => {
```

File: packages/core/src/execution-engine/node-execution-context/utils/request-helpers/__tests__/utils.test.ts
Comment (tomi) [inline packages/core/src/execution-engine/node-execution-context/utils/request-helpers/__tests__/utils.test.ts:1]: These were all moved to `axios-utils.ts`

File: packages/core/src/execution-engine/node-execution-context/utils/request-helpers/utils.ts
Comment (tomi) [inline packages/core/src/execution-engine/node-execution-context/utils/request-helpers/utils.ts:1]: These were all moved to `axios-utils.ts`

File: packages/core/src/execution-engine/node-execution-context/utils/request-helper-functions.ts [hunk 4]
```diff
@@ -77,157 +75,16 @@ import { parseIncomingMessage } from './parse-incoming-message';
...
+	generateContentLengthHeader,
+	getBeforeRedirectFn,
+	getHostFromRequestObject,
+	isIgnoreStatusErrorConfig,
+	searchForHeader,
 	setAxiosAgents,
 	tryParseUrl,
-} from './request-helpers/utils';
-
-function validateUrl(url?: string): boolean {
```
Comment (tomi) [inline packages/core/src/execution-engine/node-execution-context/utils/request-helper-functions.ts:85]: This was removed, we only have `tryParseUrl` now

File: packages/core/src/execution-engine/node-execution-context/utils/request-helper-functions.ts [hunk 5]
```diff
@@ -77,157 +75,16 @@ import { parseIncomingMessage } from './parse-incoming-message';
...
-	if (!url) return false;
-
-	return tryParseUrl(url) !== null;
-}
-
-function isIgnoreStatusErrorConfig(
```
Comment (tomi) [inline packages/core/src/execution-engine/node-execution-context/utils/request-helper-functions.ts:91]: All these were moved as is, no changes to logic

File: packages/core/src/execution-engine/node-execution-context/utils/request-helpers/__tests__/axios-utils.test.ts (hunk 3 above)
Comment (tomi) [reply packages/core/src/execution-engine/node-execution-context/utils/request-helpers/__tests__/axios-utils.test.ts]: Thanks! I ended removing the `isValidUrl` altogether in favour of `tryParseUrl`

**Review by ivov — APPROVED** [review]
